All rest calls

michaelgub opened the pull request:
All REST Calls implemented with no build errors.  
Problem now is that the page is not showing anything now because the testData is no longer in the code. Need to check this.

s1lobnig commented:
It is not working because area-id and relative are query parameter and not path parameters.
Also in the apiary file the query parameter is called area instead of area-id.
Unfortunatly area=[1,2,3] will not work, we have to change this (also in the apiary file) to area=1,2,3.

michaelgub commented:
Changed now from area-id to area and switched to RequestParam now the site works and I can see the graphs again.

Now I get this Error: Failed 
to convert value of type 'java.lang.String' to required type 'int'; nested exception is java.lang.NumberFormatException: For input string: "generalSituation"]   

No idea how to fix this.

Manuel57 commented:
> Changed now from area-id to area and switched to RequestParam now the site works and I can see the graphs again.
> 
> Now I get this Error: Failed
> to convert value of type 'java.lang.String' to required type 'int'; nested exception is java.lang.NumberFormatException: For input string: "generalSituation"]
> 
> No idea how to fix this.

This error appears when you just request on `http://localhost:8080/daily/generalSituation` without any request parameter. But request parameters also do not work because you have this `?{area}`... everywhere in the path. remove everything which is behind the ? and also the ? from the paths to make it working.

michaelgub commented:
So die meisten Rest Calls funktionieren bis auf die:
Datenbankabfragen schauen eigentlich sehr gut aus aber trotzdem kommen diese Fehler...

###  fail fehler 500 : Internal Server Error fällt mir jetzt auch nichts auf...
 GET http://localhost:8080/daily/hospital/?area=1,4,5

### fail Fehler 400: Bad Request --> kann eigentlich nur von der Query abhängen...
 GET http://localhost:8080/daily/generalSituation/?area=1,2,3

### fail keine Ahnung wieso Fehler 404 not found komplett gleich wie bei deaths/cases
 GET http://localhost:8080/distribution/age-sex/cured/?area=1,2,3

### fail datenbankabfrage kontrolliert finde da keinen Fehler...  Fehler 500 : Internal Server Error ???
 GET http://localhost:8080/provinces
### fail das gleiche auch kontrolliert keinen Fehler... Fehler 404: Not Found ???
 GET http://localhost:8080/districts

michaelgub commented:
ALL REST CALLS FINISHED

Manuel57 commented on ase/src/main/java/com/ase/ase/rest/controller/BasicCovidController.java:
```diff
@@ -29,42 +29,59 @@
-
     @CrossOrigin
-    @GetMapping(value = "/hospital/{province-id}", produces = "application/json")
+    @GetMapping(value = "/new-cases/", produces = "application/json")
```
the last / should be removed everywhere. ONLY /new-cases

michaelgub replied:
If the last / is removed the REST call doesn't work I tried it.

Manuel57 replied:
I also tried it and it works because it must work because it is a standard that request parameters are passed behind a ? which is immediately after the path without a / in between. The request should look like localhost:8080/daily/new-cases?area=1,2

Manuel57 replied:
Check https://www.baeldung.com/spring-request-param - there it also works this way

Manuel57 commented on apiary.apib:
```diff
@@ -18,7 +18,7 @@ Polls is a simple API allowing consumers to view polls and vote in them.
         }
 
-## Questions Collection [/daily/hospital?area=[1,2,3]]
+## Questions Collection [/daily/hospital/?area=1,2,3]
```
There should be no / before the ?

Manuel57 replied:
in each call

Manuel57 commented on ase/src/main/java/com/ase/ase/rest/controller/DistributionCovidController.java:
```diff
@@ -1,25 +1,51 @@
-                ))
-        );
+    public ResponseEntity getSexDistributionBy(@RequestParam("area") Set<Integer> areas) {
+        String g = sexAndAgeDistributionRepository.getSexDistributionBy(areas);
```
We do not need an extra variable, just return ...ok(sesAndAge...getSexDistrBy(areas)); Do this in every method

Manuel57 commented on ase/src/main/java/com/ase/ase/rest/controller/DistributionCovidController.java:
```diff
@@ -1,25 +1,51 @@
+    @CrossOrigin
+    @GetMapping(value = "/age-sex/deaths/", produces = "application/json")
+    @ResponseBody
+    public ResponseEntity getsexAndAgeDeathDistributionBy(@RequestParam("area") Set<Integer> areas) {
```
Naming: Uppercase Sex

Manuel57 commented on ase/src/main/java/com/ase/ase/rest/controller/DistributionCovidController.java:
```diff
@@ -1,25 +1,51 @@
+    @CrossOrigin
+    @GetMapping(value = "/age-sex/cured/", produces = "application/json")
+    @ResponseBody
+    public ResponseEntity getsexAndAgeCureDistributionBy(@RequestParam("area") Set<Integer> areas) {
```
Naming: Uppercase Sex

Manuel57 commented on ase/src/main/java/com/ase/ase/rest/controller/OverviewController.java:
```diff
@@ -20,4 +22,18 @@ public ResponseEntity getOverview(){
+    @CrossOrigin
+    @GetMapping(value = "/area-info/", produces = "application/json")
+    @ResponseBody
+    public ResponseEntity getRelativeAreaInfoBy(@RequestParam("area") Set<Integer> areas, @RequestParam("relative") boolean relative){
```
Would be nice to have the relative parameter optional. By default, it should be false. You can use the field required=true in the @RequestParam. Check https://www.baeldung.com/spring-request-param for how to do it.

Manuel57 commented:
And please do not commit and push every method change. Do it -> build it locally -> try it locally -> push it.
